flake8: add page (#6275)
tldr-pages/tldr: pages/common/flake8.md

# flake8

> Tool to check the style and quality of Python code.
> More information: <https://flake8.pycqa.org/>.

> - Lint a file or directory:

Reviewer: ```suggestion
- Lint a file or directory recursively:
```
Maybe it's obvious, so only apply if you want to.

Author: Good idea! Will apply.

`flake8 {{path/to/file_or_directory}}`

> - Show source code for each error:

Reviewer: Does it only print the line or also the lines around it?

Author: It only prints the line.

Reviewer: How about
```suggestion
- Show the line in which each error occured:
```
?

Author: Good by me. Will commit.

> - Show the line in which each error occured:

Reviewer 2: Perhaps this will read better?
```suggestion
- Show the line on which each error occurred:
```

`flake8 --show-source {{path/to/file_or_directory}}`

- Ignore a list of rules:

`flake8 --ignore {{rule1,rule2}} {{path/to/file_or_directory}}`

> - Exclude files matching the given patterns:

Reviewer: Is there maybe a command to list all available rules and patterns?

Author: I can't find one to list the rules. There is a list at https://www.flake8rules.com/.
As file patterns go, flake8 uses globs and substring matches.

Reviewer: Then maybe we should be more specific here:
```suggestion
- Exclude files matching the given globs or substrings:
```
And for the rules, how about
```diff
- Ignore a list of rules:
+ Ignore a list of rules. All available rules can be found at flake8rules.com:
```

Author: Terrific! Will push changes in next commit.

> - Show the line on which each error occurred:
> 
> `flake8 --show-source {{path/to/file_or_directory}}`
> 
> - Ignore a list of rules. (All available rules can be found at flake8rules.com):
> 
> `flake8 --ignore {{rule1,rule2}} {{path/to/file_or_directory}}`
> 
> - Exclude files matching the given globs or substrings:
> 

Reviewer 3: ```suggestion
- Lint a file or directory recursively and how the line on which each error occurred:

`flake8 --show-source {{path/to/file_or_directory}}`

- Lint a file or directory recursively and ignore a list of rules. (All available rules can be found at flake8rules.com):

`flake8 --ignore {{rule1,rule2}} {{path/to/file_or_directory}}`

- Lint a file or directory recursively but exclude files matching the given globs or substrings:

```

Reviewer 3: hmm, that doesn't show changes very clearly, however I have only added the first little sentence to each since I like complete clarity.
